game.py

```diff
@@ -405,26 +478,109 @@ def _button_press_cb(self, win, event):
+                self._correct += 2
+                self._correct_for_level += 1
             else:
                 self._opts[i].set_label('☹')
-                self._correct = 0
+                self._correct_for_level = 0
 
         if self._game in [0, 1, 3]:
             for i in range(self._level):
                 self._dots[i].set_layer(100)
         else:
+            for question_shape in self._question:
+                question_shape.hide()
             for dot in self._dots:
                 dot.set_shape(self._new_dot_surface(
                         image=self._recall_list[self._answer]))
                 dot.set_layer(100)
 
-        if self._correct == 0:
-            self._timeout_id = GLib.timeout_add(5000, self.new_game)
+        if self._correct > current:
+            self._timeout_id = GLib.timeout_add(1000, self.new_game)
         else:
-            self._timeout_id = GLib.timeout_add(3000, self.new_game)
+            self._timeout_id = GLib.timeout_add(2000, self._game_over)
         return True
-
+        
+    def _game_over(self):
+        for opt in self._opts:
+            opt.hide()
+            opt.type = -1
+        for dot in self._dots:
+            dot.hide()
+        for question_shape in self._question:
+            question_shape.hide()
+        self.save_highscore()
+        yoffset = int(self._space / 4.)
+        xoffset = int((self._width - 6 * self._dot_size - \
+                                   5 * self._space) / 2.)
+        y = 1
+        i=0
+        for x in range(2,6):
+            self._gameover.append(
+                Sprite(self._sprites,
+                    xoffset + (x-0.25) * (self._dot_size - 15),
+                    y * (self._dot_size - 90+ self._space) + yoffset,
+                    self._new_dot_surface(color=self._colors[1])))
+            self._gameover[-1].type = -1  # No image
+            self._gameover[-1].set_label_attributes(72)
+            i+=1
+        text = []
+        text.append("☻")
+        text.append("  Game  ")
+        text.append("  Over  ")
+        text.append("☻")
+        i = 0
+        for x in range(4) :
+            self._gameover[x].type = -1
+            self._gameover[x].set_shape(self._new_dot_surface(
+                        self._colors[2]))
+            self._gameover[x].set_label(text[i])
+            self._gameover[x].set_layer(100)
+            i += 1
+        y = 2
+        for x in range(2,5):
+            self._score.append(
+                Sprite(self._sprites,
+                    xoffset + (x + 0.25) * (self._dot_size - 15),
+                    y * (self._dot_size -30 + self._space) + yoffset,
+                    self._new_dot_surface(color=self._colors[1])))
+            self._score[-1].type = -1  # No image
+            self._score[-1].set_label_attributes(72)
+        text = []
+        text.append("  your  ")
+        text.append(" score:  ")
+        text.append("  " + str(self._correct) + "  ")
```

Comment:
String formatting preferred 
```suggestion
        text.append(f"  {self._correct}  ")
```